docs/lang/articles/advanced/layout.md

```diff
@@ -132,177 +109,183 @@ for (int i = 0; i < 3; i++) {
 }
 ```
 
+The accessors of `x` and `y` are in reverse order for row-major arrays and colomn-major arrays, respectively. Compared with Taichi fields, there are much more code to revise when you change the memory layout.
+
 :::
 
-### Array of Structures (AoS) versus Structure of Arrays (SoA)
+<!-- ### Array of Structures (AoS) versus Structure of Arrays (SoA) -->
+### AoS versus SoA
 
-Fields of same shape can be placed together.
+AoS means array of structures and SoA means structure of arrays. Consider an RGB image with 4 pixels and 3 color channels, the AoS layout stores `RGBRGBRGBRGB` and the SoA layout stores `RRRRGGGGBBBB`.
 
-For example, the following places two 1D fields of shape `3` together, which
-is called Array of Structures (AoS):
+The selection of AoS or SoA layouts largely depends on the access pattern to the field. Let's discuss the scenario to process large RGB images. The two layouts has the following arrangement in memory:
+```
+# address: low ...................... high
+# AoS:     RGBRGBRGBRGBRGBRGB.............
+# SoA:     RRRRR...RGGGGGGG...GBBBBBBB...B
+```
+To calculate grey scale of each pixel, we need all color channels but doesn't involve other pixels. Apparently, the AoS layout has a better memory access pattern: color channels are stored continuously, and adjacent channels can be fetched instantly, while the color channels of a pixel are stored far apart in the memory space for the SoA layout.
```

Comment:
```suggestion
To calculate grey scale of each pixel, we need all color channels but do not require the value of other pixels. In this case, the AoS layout has a better memory access pattern: Since color channels are stored continuously, and adjacent channels can be fetched instantly. The SoA layout is not a good option because the color channels of a pixel are stored far apart in the memory space.
```